server/src/test/java/org/eclipse/lsp/cobol/usecases/TestFileDescriptor.java

```diff
@@ -0,0 +1,237 @@
+ *
+ */
+
+package org.eclipse.lsp.cobol.usecases;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.eclipse.lsp.cobol.service.CobolDocumentModel;
+import org.eclipse.lsp.cobol.service.CopybookProcessingMode;
+import org.eclipse.lsp.cobol.service.delegates.hover.HoverProvider;
+import org.eclipse.lsp.cobol.service.delegates.hover.VariableHover;
+import org.eclipse.lsp.cobol.service.delegates.validations.AnalysisResult;
+import org.eclipse.lsp.cobol.usecases.engine.UseCaseEngine;
+import org.eclipse.lsp4j.*;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.junit.jupiter.api.Test;
+
+import static org.eclipse.lsp.cobol.service.delegates.validations.SourceInfoLevels.ERROR;
+import static org.eclipse.lsp.cobol.service.delegates.validations.UseCaseUtils.DOCUMENT_URI;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Tests the FD and SD.
+ * 1. Hover shows the File Description and File control data.
+ * 2. FD are checked before the use of file variable.
+ */
+public class TestFileDescriptor {
+  private static final String TEXT =
+      "      *****************************************************************\n"
+          + "       IDENTIFICATION DIVISION.\n"
+          + "       PROGRAM-ID.  calc2.\n"
+          + "       ENVIRONMENT DIVISION.\n"
+          + "       INPUT-OUTPUT SECTION.\n"
+          + "       FILE-CONTROL.\n"
+          + "             SELECT {$INFILE} ASSIGN TO \"c:\\infile.dat\"\n"
+          + "             ORGANIZATION IS LINE SEQUENTIAL\n"
+          + "             ACCESS MODE IS SEQUENTIAL\n"
+          + "             FILE STATUS IS {$IFCODE}.\n"
+          + "      *****************************************************************\n"
+          + "      *****************************************************************\n"
+          + "       DATA DIVISION.\n"
+          + "       FILE SECTION.\n"
+          + "       FD  {$*INFILE}\n"
+          + "           LABEL RECORDS ARE STANDARD\n"
+          + "           DATA RECORD IS InputRec\n"
+          + "           RECORD CONTAINS 80 CHARACTERS\n"
+          + "           RECORDING MODE IS F\n"
+          + "           BLOCK CONTAINS 0 RECORDS.\n"
+          + "       01  {$*InputRec}.\n"
+          + "           05 {$*Field1} PIC X(40).\n"
+          + "           05 {$*Field2} PIC X(40).\n"
+          + "\n"
+          + "       WORKING-STORAGE SECTION.\n"
+          + "       01  {$*FILE-STATUS-CODES}.\n"
+          + "           05  {$*IFCODE}           PIC X(2).\n"
+          + "               88 {$*CODE-READ}     VALUE SPACES.\n"
+          + "               88 {$*NO-MORE-DATA}  VALUE \"10\".\n"
+          + "\n"
+          + "       PROCEDURE DIVISION.\n"
+          + "            READ {$INFILE}.\n"
+          + "            CLOSE {$INFILE}.";
+
+  private static final String HOVER =
+      "DESCRIPTION "
+          + System.lineSeparator()
```

Comment:
Isn't `\n` enough?

Reply:
To trim the spaces (for indentation on hover), I replaced [newLine + spaces] with System.lineSeparator .